CIP-0103/README.md

```diff
@@ -0,0 +1,86 @@
+
+### Error Types
+
+#### APIError
+See [CIP-30 (Cardano dApp-Wallet Web Bridge) APIError](https://cips.cardano.org/cips/cip30/#apierror)
+
+#### TxSignError
+See [CIP-30 (Cardano dApp-Wallet Web Bridge) TxSignError](https://cips.cardano.org/cips/cip30/#txsignerror)
+
+#### TxSendError
+See [CIP-30 (Cardano dApp-Wallet Web Bridge) TxSendError](https://cips.cardano.org/cips/cip30/#txsignerror)
+
+### `api.cip103.signTxs(txs: TransactionSignatureRequest[]): Promise<cbor<transaction_witness_set>[]>`
+
+Errors: `APIError`, `TxSignError`
+
+Signs a list of transactions where each transaction can either be independent and/or as a sequence of interconnected transactions where a subsequent transaction depends on a previous one. The returned array of witness sets directly correspond to the elements in the `txs` parameter, aligning the witness set at index 0 with the transaction at index 0, and so forth.
+
+On sign error for any transaction in the array, no witnesses are to be returned. Instead a `TxSignError` is to be thrown. The error message thrown should include a reference to the transaction that caused the sign error, by including the index for failing transaction that map to the input array transaction list.
+
+There are certain things that should be considered by wallets implementing this CIP, namely user visibility of what is signed. Though not explicitly specified in this CIP, as it would be up to the wallet to find a good solution, the wallet should make it clear to the user that multiple transactions are to be signed, and to give a clear overview/summary of what is signed. In addition to visibility, the wallet shall process the input transaction array in the same order as input parameter to allow transactions to be chained by accepting a previous transaction in the array to be used as input in a following transaction.
+
+### `api.cip103.submitTxs(txs: cbor<transaction>[]): Promise<hash32[]>`
+
+Errors: `APIError`, `TxSendError`
+
+Extends [CIP-30 (Cardano dApp-Wallet Web Bridge) submitTx](https://cips.cardano.org/cips/cip30/#apisubmittxtxcbortransactionpromisehash32) with the ability to submit transactions in bulk. Transactions are to be submitted in the same order as provided as inputs. All transactions provided as input should be attempted to be submitted even in case of error. If all transactions are successfully submitted, an array of transaction ids is to be returned. In the case of one or multiple `Refused | Failure`, a `TxSendError | hash32` array is thrown. Each entry in the array represents either a successful submit with `hash32` (transaction id), or in the case of `Refused | Failure`, a `TxSendError` object. In both cases, the response array directly corresponds to the elements in the `txs` parameter.
```

Comment:
Does this endpoint offer any advantage? The user is not required to confirm submission, so it effectively does not matter if the txs are being submitted one-by-one or at once.

Reply:
Its useful for transaction chaining where outputs from a previous in list is taken into account for later. It also helps as a single signature (password input for mnemonic) can be used to witness all transactions in the list if need be.